### ISSUE TITLE
ros requires node initialization in main thread

### ISSUE THREAD

**Issue by jcvasquezt** [nengo/nengo_gui#827]
to get a ros publisher running from the nengo_gui, we added a line in the nengo_gui/main.py file before the start call:

```python
    s = nengo_gui.gui.GUI(filename=args.filename,
                                       backend=args.backend)

    rospy.init_node('nengo_ros')   # this is the added line

    s.start(port=args.port, password=args.password, browser=args.browser)
```

**Comment by fmirus** [comment]
@jcvtieck thanks for documenting this. Saved me a lot of time.

**Comment by fmirus** [comment]
just for the sake of completeness: this hack (changing nengo_gui) can be avoided with [this code-snippet](https://github.com/bjkomer/nengo_ros/blob/master/publisher.py#L10-L11) by @bjkomer

**Comment by jcvtieck** [comment]
@fmirus, yes, that works to use only nengo in a ros node, but not with the nengo_gui, ?

**Comment by bjkomer** [comment]
@jcvtieck I've only tried it within `nengo_gui` so it should work there.

If you have multiple browser tabs running different models at the same time, only one of the models should call it.

**Comment by fmirus** [comment]
@jcvtieck I also tested this with the nengo_gui and it works. I assume the trick is to set the parameter `disable_signals=True`. The [rospy documentation](http://wiki.ros.org/rospy/Overview/Initialization%20and%20Shutdown) reads: 

By default, rospy registers signal handlers so that it can exit on Ctrl-C. In some code, you may wish to disable this, including:    
- You're not calling init_node() from the Python Main thread. Python only allows signals to be registered from the Main thread.
- You're running rospy within Qt, wxPython or any other GUI toolkit that has its own exit handlers.
-  You wish to have your own signal handlers by default.
